src/Hipace.H

```diff
@@ -127,14 +127,16 @@ public:
      * \param[in] bins slice indexing
      * \param[in] only_ghost whether to send only ghost particles
      */
-    void Notify (const int step, const int it, const amrex::Vector<BeamBins>& bins, bool only_ghost=false);
+    void Notify (const int step, const int it, const amrex::Vector<BeamBins>& bins={},
+                 bool only_ghost=false);
 
     /** \brief When slices sent to rank downstream, free buffer memory and make buffer nullptr
      *
      * \param[in] it current box number
      * \param[in] only_ghost whether to pack only ghost particles (or only valid particles)
+     * \param[in] only_time whether only the time is send (and no beam particles)
      */
-    void NotifyFinish (const int it=0, bool only_ghost=false);
+    void NotifyFinish (const int it=0, bool only_ghost=false, bool only_time=false);
```

Comment:
Wow getting more and more dangerous here, maybe we should refrain from optional arguments and make all of them mandatory?